docs/docs/20-usage/40-secrets.md

```diff
@@ -1,45 +1,60 @@
 # Secrets
 
-Woodpecker provides the ability to store named parameters external to the YAML configuration file, in a central secret store. These secrets can be passed to individual steps of the pipeline at runtime.
+Woodpecker provides the ability to store named variables in a central secret store.
+These secrets can be passed securely to individual pipeline steps using the `from_secret` keyword.
 
-Woodpecker provides three different levels to add secrets to your pipeline. The following list shows the priority of the different levels. If a secret is defined in multiple levels, will be used following this priorities: Repository secrets > Organization secrets > Global secrets.
+Three different levels of secrets are available.
+The following list shows the priority of these.
+If a secret is defined in multiple levels, the following precedence applies: Repository secrets > Organization secrets > Global secrets.
 
-1. **Repository secrets**: They are available to all pipelines of an repository.
-2. **Organization secrets**: They are available to all pipelines of an organization.
-3. **Global secrets**: Can be configured by an instance admin.
-   They are available to all pipelines of the **whole** Woodpecker instance and should therefore **only** be used for secrets that are allowed to be read by **all** users.
+1. **Repository secrets**: Available to all pipelines of a repository.
+1. **Organization secrets**: Available to all pipelines of an organization.
+1. **Global secrets**: Can only be set by instance admins.
+   Global secret are available to all pipelines of the **entire** Woodpecker instance and should therefore be used with caution.
+
+:::tip
+In addition to the native secret integration, external secret providers can be utilized by interacting with them directly within pipeline steps.
+Access to these providers can be configured using Woodpecker secrets, enabling the retrieval of secrets from the respective external sources.
+:::
+
+:::warning
+Woodpecker can mask secrets from its native secret store, but it cannot apply the same protection to external secrets. As a result, these external secrets may be exposed in the pipeline logs.
+:::
 
 ## Usage
 
-You can set a setting or environment value from secrets using the `from_secret` syntax.
+You can set a setting or environment value from Woodpecker secrets using the `from_secret` syntax.
 
-The example below passes a secret called `secret_token` as an environment variable that will be called `TOKEN_ENV`:
+The example below passes a secret called `SECRET_TOKEN` which will be stored in an environment variable named `TOKEN_ENV`:
```

Comment:
Might be just my preference but imho in example it would be better to keep `secret_token` lowercase as it was

Reply:
IIRC we agreed on using uppercase secret refs everywhere some time ago. Can maybe somebody else verify this or was this a dream of mine?